app/uk/gov/hmrc/securemessage/models/core/FilterTag.scala

```diff
@@ -25,8 +25,11 @@ object FilterTag {
     Json.reads[FilterTag]
   }
 
-  def parse(tagString: String): FilterTag = {
+  def parse(tagString: String): Either[String, FilterTag] = {
```

Comment:
same thing, left should be a specialised error.

Reply:
Same reply as above I think this is the correct behaviour.

Reply:
I think Oleg means Secialised is an SecureMessgeError type instead of string

Reply:
I think Oleg means like specialised type SecureMessageErrror instead of string